app/views/shared/_handsontable.html.erb

```diff
@@ -13,7 +13,7 @@
 
   edtfValidator = function (value, callback) {
     setTimeout(function(){
-      if (/^\d\d\d\d(-\d\d)?(-\d\d)?$/.test(value)) {
+      if (/^(19|20)\d{2}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/.test(value) || !value) {
```

Comment:
Can you please add a comment here what this regex do? It will be hard for other devs to understand what this does.

Or store the regex on a descriptive variable.

Reply:
I've added the comment